### DISCUSSION TITLE
wrong assumption of my location relative to a bridge?

### DISCUSSION THREAD

**Discussion by wolftune** [osmandapp/OsmAnd#11217] [General]
I had a situation where driving on a road and continuing straight, the app suddenly told me to turn right onto the street I was already on. This happened when I went under a bridge, so I'm guessing it thought I was on the bridge. Is this something I should report somewhere? Are there methods to avoid that sort of mistake?

**Comment by thompehw** [comment]
I'm no expert, but it's been a few days so here is a start!

Have you checked that the bridge is mapped correctly in OpenStreetMap? If you're not confident or interested to do that, if you are OK to give the location, then other people could take a look.

Best wishes.

**Reply by wolftune** [reply]
This was reported to me by my mother who was using OSMand, so I figured out in context what happened. I did check out the map, and yes, it's precisely correct. The issue was exclusive to the navigation, I think. I can double check that the map that is on the device is itself up to date.

**Reply by thompehw** [reply]
You could try simulating the navigation (plan the route in OsmAnd, then in the route settings pick "simulate", then start navigation). 

If that doesn't reproduce the problem, then I'd guess it was due to a GPS error? OsmAnd has a "tunnel simulation" feature, but from a casual glance at the code that looks like it triggers "tunnel simulation" ([here](https://github.com/osmandapp/OsmAnd/blob/40286f93b2509091cdaab19fd346bf0e18df0c96/OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java#L709)) , it requires both a tunnel tag and layer=-1. I guess bridges aren't normally a problem? 

Unless someone else recognises this problem, people would need a way of reproducing it to be able to help. Good luck!

**Reply by thompehw** [reply]
Looking again at that code (just out of curiosity), tunnel simulation to cope with lost GPS is triggered by tunnel=yes OR layer=-1 (not AND as I said previously). But typically a road under a bridge stays on layer=0 (default) and the bridge itself has layer=1

**Reply by wolftune** [reply]
I should be specific, the location was in the vicinity of 45.4643, -122.66883

And so, it wasn't a matter of a straightforward bridge situation but a matter of the interchange. The directions said to turn right onto OR-43 in a situation where turning right would take someone off of OR-43 and onto the bridge. That's what the report to me was at least.

It's possible that it was not the bridge particularly but confusing directions about how to navigate the bridge-access vs the bridge-underpass

At this point, I'm assuming not much can be figured out without actively reproducing the error.
